src/Libraries/Microsoft.Extensions.AI.Abstractions/Functions/AIFunctionFactoryOptions.cs

```diff
@@ -122,7 +123,15 @@ public AIFunctionFactoryOptions()
     /// be disposed of after the invocation completes.
     /// </para>
     /// </remarks>
-    public Func<Type, AIFunctionArguments, object>? CreateInstance { get; set; }
+    public CreateInstanceFunc? CreateInstance { get; set; }
+
+    /// <summary>Delegate used with <see cref="CreateInstance"/> to create an instance of <paramref name="targetType"/>.</summary>
+    /// <param name="targetType">The type to instantiate.</param>
+    /// <param name="arguments">The arguments passed to <see cref="AIFunction.InvokeAsync"/>.</param>
+    /// <returns>The instantiated object.</returns>
+    public delegate object CreateInstanceFunc(
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] Type targetType,
```

Comment:
Nit: as discussed offline, the annotation is not strictly necessary for delegates that may be hardcoding a particular set of types. It feels like it exists solely to provide dependency inversion for `ActivationUtilities` and because of linker issues when the DAM attribute is absent. 

cc @eerhardt

Reply:
> feels like it exists solely to provide dependency inversion for ActivationUtilities and because of linker issues when the DAM attribute is absent.

The property is there to avoid AIFunctionFactory depending on DI.

The delegate is there because of the attribute.